app/controllers/ems_common.rb

```diff
@@ -465,6 +465,7 @@ def form_instance_vars
     end
     @nuage_api_versions = retrieve_nuage_api_versions
     @hawkular_security_protocols = retrieve_hawkular_security_protocols
+    @disabled_ems_infra_types = ['Kubevirt', 'kubevirt']
```

Comment:
Let's move this to `ems_infra_controller` since it is specific to that as shown below -

```diff
diff --git a/app/controllers/ems_common.rb b/app/controllers/ems_common.rb
index 91e71e567..b21db677a 100644
--- a/app/controllers/ems_common.rb
+++ b/app/controllers/ems_common.rb
@@ -465,7 +465,6 @@ module EmsCommon
     end
     @nuage_api_versions = retrieve_nuage_api_versions
     @hawkular_security_protocols = retrieve_hawkular_security_protocols
-    @disabled_ems_infra_types = ['Kubevirt', 'kubevirt']
   end
 
   def retrieve_provider_regions
diff --git a/app/controllers/ems_infra_controller.rb b/app/controllers/ems_infra_controller.rb
index 181174b95..3c10d1343 100644
--- a/app/controllers/ems_infra_controller.rb
+++ b/app/controllers/ems_infra_controller.rb
@@ -35,6 +35,11 @@ class EmsInfraController < ApplicationController
     redirect_to :action => 'show_list'
   end
 
+  def new
+    @disabled_ems_infra_types = ['Kubevirt', 'kubevirt']
+    super
+  end
+
   def scaling
     assert_privileges("ems_infra_scale")
```

Reply:
@AparnaKarve I applied this change, but it causes an exception in 'Edit' infra provider:

`[----] I, [2018-01-17T22:52:54.945943 #11537:c8c731c]  INFO -- :   Rendering plugins/manageiq-ui-classic/app/views/ems_infra/edit.html.haml within layouts/application
[----] I, [2018-01-17T22:52:54.952330 #11537:c8c731c]  INFO -- :   Rendered plugins/manageiq-ui-classic/app/views/layouts/_flash_msg.html.haml (0.0ms)
[----] I, [2018-01-17T22:52:54.954595 #11537:c8c731c]  INFO -- :   Rendered plugins/manageiq-ui-classic/app/views/ems_infra/_form.html.haml (2.6ms)
[----] I, [2018-01-17T22:52:54.954922 #11537:c8c731c]  INFO -- :   Rendered plugins/manageiq-ui-classic/app/views/ems_infra/edit.html.haml within layouts/application (8.8ms)
[----] F, [2018-01-17T22:52:54.955267 #11537:c8c731c] FATAL -- : Error caught: [ActionView::Template::Error] no implicit conversion of nil into Array
/home/masayag/work/manageiq-master/plugins/manageiq-ui-classic/app/views/ems_infra/_form.html.haml:26:in `concat'
/home/masayag/work/manageiq-master/plugins/manageiq-ui-classic/app/views/ems_infra/_form.html.haml:26:in `_plugins_manageiq_ui_classic_app_views_ems_infra__form_html_haml___1155577326132250092_70308680842400'
/home/masayag/.rvm/gems/ruby-2.3.3/gems/actionview-5.0.6/lib/action_view/template.rb:159:in `block in render'
`

Reply:
```
 def edit
    @disabled_ems_infra_types = []
    super
  end
```

makes it work, but I'm not sure if that's the right way to go

Reply:
Edit Provider does not have the emstype dropdown, hence I did not suggest overriding the `edit` method in the controller in my above comment.

It looks like the emstype dropdown is considered for a render before the JS 
`"ng-if" => "newRecord"` condition kicks in, which seems to be the reason for the above error.

I think we can address the above error like this -
```diff
diff --git a/app/views/ems_infra/_form.html.haml b/app/views/ems_infra/_form.html.haml
index 0b9587918..ec9b92ddb 100644
--- a/app/views/ems_infra/_form.html.haml
+++ b/app/views/ems_infra/_form.html.haml
@@ -24,7 +24,7 @@
         = _('Type')
       .col-md-8
         = select_tag('emstype',
-                     options_for_select([["<#{_('Choose')}>", nil]] + @ems_types, disabled: ["<#{_('Choose')}>", nil].concat(@disabled_ems_infra_types)),
+                     options_for_select([["<#{_('Choose')}>", nil]] + @ems_types, disabled: ["<#{_('Choose')}>", nil].concat(@disabled_ems_infra_types || [])),
```

Reply:
The rationale behind the above change https://github.com/ManageIQ/manageiq-ui-classic/pull/3143#discussion_r162145873 is to make sure that going forward, we keep *only* the common data/code in ems_common, so that it does not get out of hand.

Reply:
Thanks for the detailed explanation.